zebra-state/src/service/finalized_state/disk_db/tests.rs

```diff
@@ -24,3 +24,25 @@ impl DiskDb {
         rocksdb::DB::list_cf(&opts, path)
     }
 }
+
+/// Check that zs_iter_opts returns an upper bound one greater than provided inclusive end bounds.
+#[test]
+fn zs_iter_opts_increments_key_by_one() {
+    let _init_guard = zebra_test::init();
+
+    let keys: [u32; 13] = [
+        0, 1, 200, 255, 256, 257, 65535, 65536, 65537, 16777215, 16777216, 16777217, 16777218,
```

Comment:
This is missing a test case for u32::MAX

Reply:
Otherwise this test has good coverage and maybe we don't need a proptest, since we know the structure of the code.

Reply:
> This is missing a test case for u32::MAX

Added in https://github.com/ZcashFoundation/zebra/pull/7731/commits/600aa5f6087dc0b6af9d9fcce54b614cedaee5c4.